src/store/index.ts

```diff
@@ -0,0 +1,63 @@
+import { configureStore, combineReducers, current } from '@reduxjs/toolkit';
+import { readFileSync, writeFile } from 'fs';
+import { version as latest, PERSISTED_STATE_FILE } from '../config';
+import zoomInfoReducer, { name as zoomInfoSlice } from './zoom';
+
+const rootReducer = combineReducers({
+	[zoomInfoSlice]: zoomInfoReducer,
+});
+
+export type RootState = ReturnType<typeof rootReducer>;
+
+// redux makes it super simple to recover state after a restart
+const loadState = (): RootState | undefined => {
+	try {
+		// if the state shape changes, there should be logic added to convert from the older format
+		const { state, version } = JSON.parse(readFileSync(PERSISTED_STATE_FILE, 'utf-8'));
+		return state;
+	} catch (e) {
+		return undefined;
+	}
+};
+
+const store = configureStore({
+	reducer: rootReducer,
+	preloadedState: loadState(),
+	devTools: false,
+});
+
+// all changes are persisted to disk
+store.subscribe(() => {
+	const state = store.getState();
+	writeFile(
+		PERSISTED_STATE_FILE,
+		JSON.stringify({
+			state,
+			version: latest,
+		}),
+		() => {},
```

Comment:
We should at least log on failure. And/or, try to ping the `@Bot Builder` role in `#debugging`.

Reply:
Bumping this

Reply:
And/or, for consistency, switch to `fs.promises.writeFile`